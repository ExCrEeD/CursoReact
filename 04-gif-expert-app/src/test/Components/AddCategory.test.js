import {shallow} from 'enzyme'
import { AddCategory } from '../../Components/AddCategory'


describe('pruebas en el componente', () => {

    const setCategories=jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });

    test('debe de renderizar el componente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('debe de cambiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'hola mundo';
        input.simulate('change',{target:{value}});
    });
    
    test('no debe de postear la informacion con submit ', () => {
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    });
    
    test('debe de llamar el setCategories y limpiar la caja de text', () => {
        const input = wrapper.find('input');
        const value = 'hola mundo';
        input.simulate('change',{target:{value}});
        wrapper.find('form').simulate('submit',{preventDefault(){}});
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect(input.prop('value')).toBe('')
    })
    
});
