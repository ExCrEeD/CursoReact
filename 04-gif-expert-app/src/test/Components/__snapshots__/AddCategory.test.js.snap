// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente debe de renderizar el componente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
